internal/installcfg/installcfg.go

```diff
@@ -292,8 +277,9 @@ func (i *installConfigBuilder) setBMPlatformInstallconfig(cluster *common.Cluste
 		if swag.StringValue(host.Status) == models.HostStatusDisabled {
 			continue
 		}
-		i.log.Infof("host name is %s", hostutil.GetHostnameForMsg(host))
-		hosts[yamlHostIdx].Name = i.getBMHName(host, &masterIdx, &workerIdx)
+		hostName := hostutil.GetHostnameForMsg(host)
+		i.log.Infof("host name is %s", hostName)
+		hosts[yamlHostIdx].Name = hostName
```

Comment:
Just curious, is there any validation that checks that all hosts have different hostnames (I guess there isn't).
So far each BMH had a unique Name, what will happen if 2 (or more) hosts have the same hostname?

Reply:
From my tests of this PR here, the new naming schema generates `test-infra-cluster-d63379ba-master-0` and so on, i.e. `clusterName-master|worker-0..n`

I couldn't backtrace exactly why it happens like this, but my best bet is that this is because at the moment when the `install-config.yaml` is generated not all the values are available for this function... In the `GetHostnameForMsg` we are getting in order the following: either `host.RequestedHostname`, `""` (which fails back to `host.ID`) or `inventory.Hostname`.

Having said this, the CR `Nodes` is already using the name coming directly from the target machine (so either OS visible hostname or libvirt domain name). If there were duplicated hostnames, `Nodes` CR would be the first one to fail and it would have been already failing in the past. By this line of reasoning I believe that even though we don't check explicitly for duplicates, this PR here is not introducing another failure path.

Reply:
This matches my understanding too!